exercises/flatten-array/FlattenArrayExample.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+func constructList(elements: Any?...) -> [Any?] {
+
+    return elements
+}
+
+func flattenArray<T>(_ list: [Any?]) -> [T] {
+
+    var flattenedArray = [T]()
+
+    func extractArrayElements(array: [Any?]) {
+
+        for element in array {
+
+            if let j = element {
```

Comment:
What if `element` is nil... I don't see an `else` block...

Perhaps we could use `flatMap` on the array above to explicitly drop the `nil`s and only operate on items which exist? this would also reduce one level of nesting. 

`for element in array.flatmap{$0}` now every element we iterate over is known to be not `nil`.